src/core/service.c

```diff
@@ -1358,10 +1372,77 @@ static int service_collect_fds(
                 rfd_names[n_fds] = NULL;
         }
 
+        if (s->open_files) {
+                size_t n_fds = rn_socket_fds + s->n_fd_store;
+
+                LIST_FOREACH(open_files, f, s->open_files) {
+                        struct stat st;
+                        int fd = -1;
```

Comment:
This should be marked with `_cleanup_close_`

Reply:
If I understand correctly how  `_cleanup_close_` works, it wil close the file at the exit of the function.

I don't think that's what we want here since we want to pass the opened file to the service.